cdtb/tools.py

```diff
@@ -78,3 +80,11 @@ def raw(self, length):
     def unpack_string(self):
         """Unpack string prefixed by its 32-bit length"""
         return self.f.read(self.unpack('<L')[0]).decode('utf-8')
+
+def stringtable_paths(base_dir):
+    # Support all known variations of stringtable paths here:
+    # Old: 'font_config_*.txt'
+    # Newer: 'main_*.stringtable'
+    # Current: '*/main.stringtable'
+
+    return {re.search(r"(.._..)[\\/]main\.stringtable$", path).group(1): path for path in glob.glob(os.path.join(base_dir, "??_??/main.stringtable"))} or {re.search(r"main_(.._..)\.stringtable$", path).group(1): path for path in glob.glob(os.path.join(base_dir, "main_??_??.stringtable"))} or {re.search(r"fontconfig_(.._..)\.txt$", path).group(1): path for path in glob.glob(os.path.join(base_dir, "fontconfig_??_??.txt"))}
```

Comment:
This is unreadable. :/ Spread over multiple lines or use intermediate variables.